lib/elixir/lib/integer.ex

```diff
@@ -400,17 +400,23 @@ defmodule Integer do
       iex> Integer.gcd(10, 0)
       10
 
+      iex> Integer.gcd(7, 7)
+      7
+
       iex> Integer.gcd(0, 0)
       0
 
   """
   @spec gcd(integer, integer) :: pos_integer
+  @spec gcd(int, int) :: int when int: pos_integer
+  @spec gcd(0, 0) :: 0
```

Comment:
This should raise an ArithmeticError instead of returning '0'. There is no greatest common divisor of `0` and `0`, because we cannot divide by `0`.